kombu/asynchronous/semaphore.py

```diff
@@ -1,9 +1,15 @@
 """Semaphores and concurrency primitives."""
 
 from collections import deque
+from typing import TYPE_CHECKING, Callable, ParamSpec
```

Comment:
ParamSpec was introduced in 3.10, so typing_extensions is needed for the other versions.

Reply:
Should I then also put this inside the `TYPE_CHECKING` condition, in case the package is run in a runtime that doesn't have `typing_extensions` installed? I see it's not listed in the package dependencies, and it isn't built-in is it?

Reply:
Added 76e6c7a7607bb3c5b87e6983c5ed0335ae38015b as an initial change

Reply:
I guess we'd want to include the package as dependency, but only for the versions that is not supported (with a marker). And in code we need to check the version with sys.version_info